src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -650,6 +662,55 @@ public Collection<NodeProvisioner.PlannedNode> provision(
         return Collections.emptyList();
     }
 
+    private static void checkKubernetesUrlFIPS(String url) {
+        if (!FIPS140.useCompliantAlgorithms() || StringUtils.isBlank(url)) {
+            return;
+        }
+        if (!url.startsWith("https:")) {
+            throw new IllegalArgumentException(Messages.KubernetesCloud_kubernetesServerUrlIsNotSecure());
+        }
+    }
+
+    private static void checkSkipTlsVerify(boolean skipTlsVerify) {
+        if (FIPS140.useCompliantAlgorithms() && skipTlsVerify) {
+            throw new IllegalArgumentException(Messages.KubernetesCloud_skipTlsVerifyNotAllowedInFIPSMode());
+        }
+    }
+
+    private static void checkServerCertificateFIPS(String serverCertificate) {
+        if (!FIPS140.useCompliantAlgorithms()) {
+            return;
+        }
+        if (StringUtils.isBlank(serverCertificate)) {
+            throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeyEmpty());
+        }
+        try {
+            PEMEncodable pem = PEMEncodable.decode(serverCertificate);
+            Certificate cert = pem.toCertificate();
+            if (cert == null) {
+                throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateNotACertificate());
+            }
+            PublicKey publicKey = cert.getPublicKey();
+            if (publicKey instanceof RSAPublicKey) {
+                if (((RSAPublicKey) publicKey).getModulus().bitLength() < 2048) {
+                    throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeySize());
+                }
+            } else if (publicKey instanceof DSAPublicKey) {
+                if (((DSAPublicKey) publicKey).getParams().getP().bitLength() < 2048) {
+                    throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeySize());
+                }
+            } else if (publicKey instanceof ECPublicKey) {
+                if (((ECPublicKey) publicKey).getParams().getCurve().getField().getFieldSize() < 224) {
+                    throw new IllegalArgumentException(Messages.KubernetesCloud_serverCertificateKeySizeEC());
+                }
+            }
+
+        } catch (DecoderException | UnrecoverableKeyException | IOException e) {
+            // TODO: DecoderException could be removed once jdk11 support is removed
+            throw new IllegalArgumentException(e);
```

Comment:
I think this would "loose" the message (ie if you catch this then e.getMessage() will be `null` and you will have to look at e.getCause().getMessage() to get any useable details?

```
throw new IllegalArgumentException(e.getMessage(), e);
```